nilearn/_utils/testing.py

```diff
@@ -60,6 +61,47 @@ def assert_warns(warning_class, func, *args, **kw):
         return output
 
 
+# we use memory_profiler library for memory consumption checks
+try:
+    from memory_profiler import memory_usage
+
+    def with_memory_usage(func):
+        """A decorator to skip tests requiring memory_profiler."""
+        return func
+
+    def memory_used(func, *args, **kwargs):
+        """Compute memory usage when executing func."""
+        gc.collect()
+        mem_use = memory_usage((func, args, kwargs),
+                               interval=.001, timeout=1)
```

Comment:
Quickly looking at the memory_profiler code, it looks like timeout does nothing when memory usage is called with a tuple (func, args, kwargs).